comm/src/main/scala/coop/rchain/catscontrib/Kvs.scala

```diff
@@ -21,6 +21,25 @@ object Kvs extends KvsInstances {
       def put(k: K, v: V) = K.put(k, v).liftM[T]
       def delete(k: K)    = K.delete(k).liftM[T]
     }
+
+  class InMemoryKvs[F[_]: Capture, K, V] extends Kvs[F, K, V] {
+    var mem: Map[K, V] = Map.empty[K, V]
+
+    def keys: F[Vector[K]] = Capture[F].capture {
+      mem.keys.toVector
+    }
+    def get(k: K): F[Option[V]] = Capture[F].capture {
+      mem.get(k)
+    }
+    def put(k: K, v: V): F[Unit] = Capture[F].capture {
+      mem = mem + (k -> v)
```

Comment:
Does capture magically serialize these mutations?

Reply:
Capture will do whatever instance for it will do. We have an instance for `Id` that eagerly evaluates itself.

Reply:
Right, but I meant that an instance of this thing will race itself to mutate the map, won't it? My question is whether the `Capture` business helps there, or if this needs some synchronization.

Reply:
I will add concurrent, thread local map